src/main/java/com/microsoft/azure/util/AzureCredentials.java

```diff
@@ -140,7 +140,7 @@ public ServicePrincipal(
             this.clientSecret = Secret.fromString(clientSecret);
             this.oauth2TokenEndpoint = Secret.fromString(oauth2TokenEndpoint);
             this.tenant = Secret.fromString(
-                    ServicePrincipal.getTenantFromTokenEndpoint(oauth2TokenEndpoint));
+                    ServicePrincipal.getTenantFromTokenEndpoint(this.oauth2TokenEndpoint.getPlainText()));
```

Comment:
Nice catch. This happens while updating an existing service principal. If you create a new ServicePrincipal the oauth2TokenEndpoint will be in plain text.
I think the plugin should be release after this is merged & tested